dvc/fs/__init__.py

```diff
@@ -124,6 +124,9 @@ def get_cloud_fs(repo, **kwargs):
         # For WebDAVFileSystem, provided url is the base path itself, so it
         # should be treated as being a root path.
         fs_path = cls.root_marker
+    if issubclass(cls, GSFileSystem):
+        # For GSFileSystem there's no need to strip
+        fs_path = url
```

Comment:
Hm, could you explain why we are not modifying _stip_protocol instead?

Reply:
This was initially added due to a failing API-related test. It's no longer necessary, so only the https://github.com/iterative/dvc-objects/pulls/41 is required